cypress/integration/ui/filter_restrict/filter_restrict.js

```diff
@@ -0,0 +1,171 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps'
+import { dashboardChipSel } from '../../../selectors/viewDashboard'
+import { filterDimensionsPanelSel } from '../../../selectors/dashboardFilter'
+
+const TEST_DASHBOARD_PREFIX = 'TEST_FILTER_SETTINGS'
+const TEST_DASHBOARD_TITLE = `${TEST_DASHBOARD_PREFIX}-${new Date().toUTCString()}`
+
+const toggleShowMoreButton = () => {
+    cy.get('[data-test="showmore-button"]').click()
+}
+
+/*
+Background
+*/
+
+before(() => {
+    cy.request(
+        `http://localhost:8080/api/dashboards?paging=false&fields=id&filter=name:like:${TEST_DASHBOARD_PREFIX}`
+    ).then(resp => {
+        try {
+            resp.body.dashboards.forEach(dashboard => {
+                cy.request(
+                    'DELETE',
+                    `http://localhost:8080/api/dashboards/${dashboard.id}`
+                )
+            })
+        } catch (e) {
+            console.log(`${TEST_DASHBOARD_PREFIX} dashboards not deleted`)
+        }
+    })
+})
+
+/*
+Scenario: I create a new dashboard and have no Filter Restrictions
+*/
+
+When('I add a dashboard title', () => {
+    cy.get('[data-test="dashboard-title-input"]').type(TEST_DASHBOARD_TITLE)
+})
+
+When('I click on Filter Settings', () => {
+    cy.get('button').contains('Filter Settings').click()
+})
+
+Then('Filter Settings are not restricted, and I can save the dashboard', () => {
+    cy.get('[name="radio-allow-filtering-on-all"]').should('be.checked')
```

Comment:
Instead of using the name property, I'm wondering if its possible to get the element based on the label https://docs.cypress.io/api/commands/contains.html#Favor-of-lt-label-gt-over-other-deeper-elements
This is in sticking with the principle of running the test according to what the user sees (as much as possible anyway).

Reply:
fixed. All of the interactions with the radio buttons have been updated to be retrieved based on the label and then find the corresponding input